### PR TITLE
[docs] Document metrics

### LINKED ISSUE #1060: Document application metrics

### PR DESCRIPTION
**Pull request by bmorelli25**
* ~WIP~
* ~Needs https://github.com/elastic/kibana/pull/28621 to be merged first.~

Adds metrics documentation, and makes the APM Data Model easier to understand. Closes elastic/apm-server#1060

### CODE REVIEW

File: docs/guide/apm-data-model.asciidoc [hunk 1]
```diff
@@ -1,59 +1,33 @@
 [[apm-data-model]]
 == Data Model
 
-Elastic APM agents capture different types of information from within their instrumented applications - namely `transactions`, `spans`, and `errors`. 
+Elastic APM agents capture different types of information from within their instrumented applications - namely `spans`, `transactions`, `errors`, and `metricsets`.
```
Comment (nehaduggal) [inline docs/guide/apm-data-model.asciidoc]: Metricsets is not very intuitive. Should we just call it metrics?

Comment (bmorelli25) [reply docs/guide/apm-data-model.asciidoc]: Good point. I'm looking back at the design document, and it actually says they should be called "application metrics" in documentation, but that may be out of date. Tagging the team that worked on the design doc: @axw @jalvz or @graphaelli. Thoughts on calling it `metrics` vs `metricsets` vs `application metrics` in documentation?

Comment (axw) [reply docs/guide/apm-data-model.asciidoc]: +1 on "metrics"; "metricsets" is an implementation detail.

Comment (jalvz) [reply docs/guide/apm-data-model.asciidoc]: Generally agreed, I'm fine either case. We however document implementation details (ie. Intake API), in which it is called `metricsets`.

Comment (bmorelli25) [reply docs/guide/apm-data-model.asciidoc]: Great, will make the change. Thank you!

**Comment by bmorelli25** [comment]
Updated docs from `metricsets` to `metrics` per feedback.

File: docs/guide/apm-data-model.asciidoc [hunk 2]
```diff
@@ -90,24 +58,79 @@ Unforseen errors may cause spans to go missing.
 Because the agent notifies the server about how many spans there should be,
 the number of missing spans is able to be calculated and shown in the UI.
 
+[[transactions]]
+=== Transactions
```
Comment (jalvz) [inline docs/guide/apm-data-model.asciidoc]: interesting that you move transactions after spans (so you can say they are a special kind of span, i guess)
nitpick: i'd say extra or additional attributes, not `metadata` (we are overusing a bit this term)

Comment (bmorelli25) [reply docs/guide/apm-data-model.asciidoc]: @nehaduggal - curious to hear your thoughts on this

**Comment by watson** [comment]
It's not clear to me after this update that we require the metric event object to be named `metricset`. I would expect it to be documented here:

https://www.elastic.co/guide/en/apm/server/master/metricset-api.html

**Pull request merged**